cabal-install/Distribution/Client/CmdUpdate.hs

```diff
@@ -0,0 +1,179 @@
+{-# LANGUAGE CPP, NamedFieldPuns, RecordWildCards, ViewPatterns #-}
+
+-- | cabal-install CLI command: update
+--
+module Distribution.Client.CmdUpdate (
+    updateCommand,
+    updateAction,
+  ) where
+
+import Distribution.Client.ProjectOrchestration
+import Distribution.Client.ProjectConfig
+         ( ProjectConfig(..)
+         , projectConfigWithSolverRepoContext )
+import Distribution.Client.Types
+         ( Repo(..), RemoteRepo(..), isRepoRemote )
+import Distribution.Client.HttpUtils
+         ( DownloadResult(..) )
+import Distribution.Client.FetchUtils
+         ( downloadIndex )
+import Distribution.Client.JobControl
+         ( newParallelJobControl, spawnJob, collectJob )
+import Distribution.Client.Setup
+         ( GlobalFlags, ConfigFlags(..), ConfigExFlags, InstallFlags, UpdateFlags (updateIndexState)
+         , applyFlagDefaults, defaultUpdateFlags, RepoContext(..) )
+import Distribution.Simple.Setup
+         ( HaddockFlags, fromFlagOrDefault, fromFlag )
+import Distribution.Simple.Utils
+         ( die', notice, wrapText, writeFileAtomic, noticeNoWrap, intercalate )
+import Distribution.Verbosity
+         ( Verbosity, normal, lessVerbose )
+import Distribution.Client.IndexUtils.Timestamp
+import Distribution.Client.IndexUtils
+         ( updateRepoIndexCache, Index(..), writeIndexTimestamp
+         , currentIndexTimestamp )
+import Distribution.Text
+         ( Text(..), display, simpleParse )
+
+import qualified Distribution.Compat.ReadP  as ReadP
+import qualified Text.PrettyPrint          as Disp
+
+import Control.Monad (unless, when)
+import qualified Data.ByteString.Lazy       as BS
+import Distribution.Client.GZipUtils (maybeDecompress)
+import System.FilePath (dropExtension)
+import Data.Time (getCurrentTime)
+import Distribution.Simple.Command
+         ( CommandUI(..), usageAlternatives )
+import qualified Distribution.Client.Setup as Client
+
+import qualified Hackage.Security.Client as Sec
+
+updateCommand :: CommandUI (ConfigFlags, ConfigExFlags, InstallFlags, HaddockFlags)
+updateCommand = Client.installCommand {
+  commandName         = "new-update",
+  commandSynopsis     = "Updates list of known packages.",
+  commandUsage        = usageAlternatives "new-update" [ "[FLAGS]" ],
```

Comment:
I think `[FLAGS]` should reflect that we may also supply a list of repo labels. C.f.

```
Usage: cabal new-build [TARGETS] [FLAGS]
```

vs.

```
Usage: cabal new-update [FLAGS]
```

which would be better as

```
Usage: cabal new-update [FLAGS] [REPOS]
```